src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -158,6 +161,24 @@ void Thread::DisablePreemptiveMode()
         WaitForGC(m_pDeferredTransitionFrame);
     }
 }
+
+// Setup the m_pDeferredTransitionFrame field for GC helpers entered from native helper thread
+// code (e.g. ETW or EventPipe threads). Do not use anywhere else.
+void Thread::SetDeferredTransitionFrameForNativeHelperThread()
+{
+    ASSERT(ThreadStore::GetCurrentThread() == this);
+    ASSERT(!Thread::IsCurrentThreadInCooperativeMode());
+    m_pDeferredTransitionFrame = m_pTransitionFrame;
+
+    // This function can only be used when there is no managed code anywhere on the current stack (i.e., when
+    // the current thread has the simplest possible transition frame configuration). A real transition (and
+    // the implied possibility of ancestor managed code on this thread) can be deferred only in the restricted
+    // set of cases which are allowed to use DeferTransitionFrame or SetDeferredTransitionFrame.
+    if (m_pDeferredTransitionFrame != TOP_OF_STACK_MARKER)
```

Comment:
This basically assumes that the entry point like `ETW::GCLog::ForceGCForDiagnostics` cannot be reached by somebody PInvoking ETW helpers.